transport-native-io_uring/src/main/java/io/netty/channel/uring/IOUringEventLoop.java

```diff
@@ -97,27 +99,13 @@ public void run() {
                 : PlatformDependent.<Runnable>newMpscQueue(maxPendingTasks);
     }
 
-    public void add(AbstractIOUringChannel ch) {
+    void add(AbstractIOUringChannel ch) {
         logger.trace("Add Channel: {} ", ch.socket.intValue());
         int fd = ch.socket.intValue();
 
         channels.put(fd, ch);
     }
 
-    public void remove(AbstractIOUringChannel ch) {
```

Comment:
Hmm I see this is unused right now but it probably _should_ be used. It doesn't look like we currently ever remove any channels from the `channels` map ... this should be done when they are deregistered from the EL (probably after the corresponding CQEs are processed).

Reply:
it is removed: https://github.com/netty/netty/blob/267913d4a993c0309c79a53e05a4f26db4d0a3d8/transport-native-io_uring/src/main/java/io/netty/channel/uring/IOUringEventLoop.java#L238

Reply:
oh yeah we only remove channels when a poll remove is submitted https://github.com/netty/netty/blob/io-uring/transport-native-io_uring/src/main/java/io/netty/channel/uring/IOUringEventLoop.java#L254

Reply:
a poll remove event is submitted when a channel should be closed, I think we should remove channels from the channels map when no poll event is scheduled

Reply:
@normanmaurer my bad, I somehow forgot/missed the `remove` there